### PR TITLE
CRM-13408 - Add Campaign API permissions

### PR DESCRIPTION
**Pull request by jitendrapurohit**
https://issues.civicrm.org/jira/browse/CRM-13408

### CODE REVIEW

File: CRM/Core/DAO/permissions.php [hunk 1]
```diff
@@ -177,6 +177,15 @@ function _civicrm_api3_permissions($entity, $action, &$params) {
     ),
   );
 
+  // Campaign permissions
+  $permissions['campaign'] = array(
+    'default' => array(
+      'access CiviCRM',
+      'manage campaign',
+    ),
```
Comment (colemanw) [inline CRM/Core/DAO/permissions.php]: `default` means for all actions not specified. I think we should leave the default alone and target the actions we want to specify. I'd suggest:

    $permissions['campaign'] = array(
      'get' => array('access CiviCRM'),
      'create' => array(array('administer CiviCRM', 'administer CiviCampaign', 'manage campaign')),
      'update' => array(array('administer CiviCRM', 'administer CiviCampaign', 'manage campaign')),
      'delete' => array(array('administer CiviCRM', 'administer CiviCampaign')),
    );

Note the nested arrays translates to 'OR'. Some of these are iffy to me (like should 'manage campaign' give you permission to delete?) so maybe double-check the behavior in the rest of CiviCRM to verify

Comment (jitendrapurohit) [inline CRM/Core/DAO/permissions.php]: IMHO, users having only 'administer CiviCRM' should not be able to operate Campaign stuffs (Create/Delete them). Also, when I check from UI perspective, user with 'manage campaign' is having a permission to delete. Updated this PR to add these changes. Thanks for the suggestion:)

**Comment by colemanw** [comment]
looks good

**Comment by monishdeb** [comment]
test build failure not related

**Pull request merged**